### PR TITLE
feat: Added custom event event type for web analytics

### PR DESCRIPTION
**Pull request by TeddyCr**
### Describe your changes :
Implement `customEvent` event type for web analytics. The event is structured as follow

```json
{
  "eventId": "26c2ad54-ea0e-4caf-8be7-9ece0fd10de3",
  "eventData": {
    "url": "/explore/dashboards",
    "fullUrl": "http://localhost:8585/explore/dashboards",
    "hostname": "localhost",
    "eventType": "CLICK",
    "sessionId": "ae90939c-328a-4534-9ee6-d5dc3d9098fc",
    "eventValue": "Orders Dashboard"
  },
  "eventType": "CustomEvent",
  "timestamp": 1678716564284
}
```

where `eventType` is of type `CustomEventTypes.CLICK`, `sessionId` should match that of the one passed to the `pageView` for that user, `eventValue` is the text value of the html element (e.g. `<a>my button text</a>` would set `eventValue` to "my button text").

#
### Type of change :

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by TeddyCr** [comment]
cc @Sachin-chaurasiya @sureshms @harshach for feedback on the info captured.

File: openmetadata-spec/src/main/resources/json/schema/analytics/webAnalyticEventType/customEvent.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{
+  "$id": "https://open-metadata.org/schema/analytics/customEvent.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "pageViewData",
+  "type": "object",
+  "javaType": "org.openmetadata.schema.analytics.customEvent",
```
Comment (nahuelverdugo) [inline openmetadata-spec/src/main/resources/json/schema/analytics/webAnalyticEventType/customEvent.json:6]:
```suggestion
  "javaType": "org.openmetadata.schema.analytics.CustomEvent",
```
Java Class name convention: should be nouns, in mixed cases with the first letter of each internal word capitalized.

File: openmetadata-spec/src/main/resources/json/schema/analytics/webAnalyticEventType/customEvent.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        },
+        "eventType": {
+          "description": "Type of event that was performed",
+          "$ref": "#/definitions/customEventTypes"
+        },
+        "eventValue": {
+            "description": "Value of the event",
+            "type": "string"
+            }
+        }
```
Comment (nahuelverdugo) [inline openmetadata-spec/src/main/resources/json/schema/analytics/webAnalyticEventType/customEvent.json:42]: nit: 2 spaces for indenting

Comment (nahuelverdugo) [reply openmetadata-spec/src/main/resources/json/schema/analytics/webAnalyticEventType/customEvent.json:42]: I think there are common properties with `PageViewEvent.json` that could be part of `webAnalyticEventData.json` like fullUrl, url, hostname, sessionId, etc.

**Review by nahuelverdugo — APPROVED** [review]

**Pull request merged**